### ISSUE TITLE
Support monaco editor webpack plugin

### ISSUE THREAD

**Issue by zoolsher** [web-infra-dev/rspack#2890]
### What problem does this feature solve?

My web application is building with `monaco-editor-webpack-plugin` which doesn't work with rspack.

Link: https://github.com/microsoft/monaco-editor/blob/main/webpack-plugin/src/plugins/AddWorkerEntryPointPlugin.ts

### What does the proposed API of configuration look like?

IMHO. This plugin requires API like `childCompiler`, `LoaderTargetPlugin`, etc.

**Comment by fuergaosi233** [comment]
I have the same problem 🙏

**Comment by hardfist** [comment]
blocked by https://github.com/web-infra-dev/rspack/issues/1320

**Comment by ahabhgk** [comment]
You don't need the plugin if you use worker, check out example/monaco-editor-ts-react

**Comment by hardfist** [comment]
@zoolsher please help check whether native worker supports fixes your issue?

**Comment by suxiaoshao** [comment]
> You don't need the plugin if you use worker, check out example/monaco-editor-ts-react



I use examples inside the monaco example, there are still some problems (such as clicking on the keyboard to delete the key will be deleted more)

**Comment by hardfist** [comment]
> > You don't need the plugin if you use worker, check out example/monaco-editor-ts-react
> 
> I use examples inside the monaco example, there are still some problems (such as clicking on the keyboard to delete the key will be deleted more)

can you file a separate issue or add more detailed information here
